File: apps/api/src/modules/timeline/timeline.service.ts
import { Inject, Injectable } from "@nestjs/common";
import type { TimelinePageDto } from "@today/contracts";
import { CheckinService } from "../checkin/checkin.service";
import { SummaryService } from "../summary/summary.service";

/** timeline — 只读聚合，不写业务数据 */
@Injectable()
export class TimelineService {
  constructor(
    @Inject(CheckinService) private readonly checkins: CheckinService,
    @Inject(SummaryService) private readonly summaries: SummaryService,
  ) {}

  list(limit = 30): TimelinePageDto {
    const items = this.checkins.listRecent(limit).map((c) => {
      try {
        const s = this.summaries.getByDate(c.date);
        return {
          date: c.date,
          highlight: s.highlight,
          oneLiner: s.oneLiner,
          mood: s.mood,
          moodLabel: s.moodLabel,
          checkinId: c.id,
        };
      } catch {
        return {
          date: c.date,
          highlight: c.rawText.slice(0, 28) || "留下了今天",
          oneLiner: c.rawText,
          mood: "okay" as const,
          moodLabel: "平常",
          checkinId: c.id,
        };
      }
    });

    return { items, nextCursor: null };
  }
}

File: apps/api/src/main/java/com/today/timeline/TimelineService.java
package com.today.timeline;

import com.today.checkin.CheckinDto;
import com.today.checkin.CheckinService;
import com.today.common.Mood;
import com.today.summary.DaySummaryDto;
import com.today.summary.SummaryService;
import java.util.List;
import org.springframework.stereotype.Service;

@Service
public class TimelineService {

  private final CheckinService checkins;
  private final SummaryService summaries;

  public TimelineService(CheckinService checkins, SummaryService summaries) {
    this.checkins = checkins;
    this.summaries = summaries;
  }

  public TimelinePageDto list(int limit) {
    List<TimelineItemDto> items =
        checkins.listRecent(limit).stream().map(this::toItem).toList();
    return new TimelinePageDto(items, null);
  }

  private TimelineItemDto toItem(CheckinDto checkin) {
    try {
      DaySummaryDto s = summaries.getByDate(checkin.date());
      return new TimelineItemDto(
          checkin.date(),
          s.highlight(),
          s.oneLiner(),
          s.mood(),
          s.moodLabel(),
          checkin.id());
    } catch (Exception e) {
      String raw = checkin.rawText();
      String highlight = raw.length() > 28 ? raw.substring(0, 28) : raw;
      if (highlight.isBlank()) {
        highlight = "留下了今天";
      }
      return new TimelineItemDto(
          checkin.date(), highlight, raw, Mood.okay, "平常", checkin.id());
    }
  }
}
